lib/worker.js

```diff
@@ -0,0 +1,185 @@
+'use babel';
+
+/* global emit */
+
+import path from 'path';
+
+process.title = 'linter-tslint worker';
+
+const TSLINT_MODULE_NAME = 'tslint';
+let tslintDef;
+let requireResolve;
+const tslintCache = new Map();
+const config = {
+  useLocalTslint: false,
+};
+
+/**
+ * Shim for TSLint v3 interoperability
+ * @param {Function} Linter TSLint v3 linter
+ * @return {Function} TSLint v4-compatible linter
+ */
+function shim(Linter) {
+  function LinterShim(options) {
+    this.options = options;
+    this.results = {};
+  }
+
+  // Assign class properties
+  Object.assign(LinterShim, Linter);
+
+  // Assign instance methods
+  LinterShim.prototype = Object.assign({}, Linter.prototype, {
+    lint(filePath, text, configuration) {
+      const options = Object.assign({}, this.options, { configuration });
+      const linter = new Linter(filePath, text, options);
+      this.results = linter.lint();
+    },
+    getResult() {
+      return this.results;
+    },
+  });
+
+  return LinterShim;
+}
+
+function loadDefaultTSLint() {
+  if (!tslintDef) {
+    tslintDef = require('loophole').allowUnsafeNewFunction(() =>
```

Comment:
Do we actually need the loophole here? I'm not sure that Atom's CSP actually applies to Task processes.